src/app/features/charts/components/score-barchart/score-barchart.component.ts

```diff
@@ -121,22 +132,29 @@ export class ScoreBarChartComponent implements OnChanges, AfterViewInit, OnDestr
     svg.style('display', 'block');
   }
 
+  getChartBoundingWidth(): number {
+    return (
+      d3.select(this.chartRef.nativeElement).node()?.getBoundingClientRect()
+        .width || this.MIN_CHART_WIDTH
+    );
+  }
+
   createChart() {
     this.initData();
     if (this.chartData) {
-      const width = 350;
+      const width = this.getChartBoundingWidth();
       const height = 350;
```

Comment:
Perhaps we also pull height out of this method, so we have all of our variables at the top?

Reply:
Sure! Done in the next commit